probe/awsecs/client.go

```diff
@@ -106,45 +216,173 @@ func (c ecsClient) getTasks(taskArns []string) (map[string]*ecs.Task, error) {
 		Tasks:   taskPtrs,
 	})
 	if err != nil {
-		return nil, err
+		log.Warnf("Failed to describe ECS tasks, ECS service report may be incomplete: %v", err)
+		return
 	}
 
 	for _, failure := range resp.Failures {
 		log.Warnf("Failed to describe ECS task %s, ECS service report may be incomplete: %s", *failure.Arn, *failure.Reason)
 	}
 
-	results := make(map[string]*ecs.Task, len(resp.Tasks))
 	for _, task := range resp.Tasks {
-		results[*task.TaskArn] = task
+		c.taskCache.Set(*task.TaskArn, newECSTask(task))
+	}
+}
+
+// Try to match a list of task ARNs to service names using cached info.
+// Returns (task to service map, unmatched tasks). Ignores tasks whose startedby values
+// don't appear to point to a service.
+func (c ecsClientImpl) matchTasksServices(taskARNs []string) (map[string]string, []string) {
+	const servicePrefix = "ecs-svc"
+
+	deploymentMap := map[string]string{}
+	for _, serviceNameRaw := range c.serviceCache.Keys() {
+		serviceRaw, err := c.serviceCache.Get(serviceNameRaw)
+		if err != nil {
+			// This is rare, but possible if service was evicted after the loop began
+			continue
+		}
+		serviceName := serviceNameRaw.(string)
+		service := serviceRaw.(EcsService)
+		for _, deployment := range service.DeploymentIDs {
+			deploymentMap[deployment] = serviceName
+		}
+	}
+	log.Debugf("Mapped %d deployments from %d services", len(deploymentMap), c.serviceCache.Len())
+
+	results := map[string]string{}
+	unmatched := []string{}
+	for _, taskARN := range taskARNs {
+		taskRaw, err := c.taskCache.Get(taskARN)
+		if err != nil {
+			// this can happen if we have a failure while describing tasks, just pretend the task doesn't exist
+			continue
+		}
+		task := taskRaw.(EcsTask)
+		if !strings.HasPrefix(task.StartedBy, servicePrefix) {
```

Comment:
My instinct is to try to keep the data normalized, ie. just store raw fields and not derived ones. But I could do something like `ecsTask.isServiceManaged()` so that the check itself is in one obvious place (and it's not exactly an expensive check).